File: bar_frontend/src/app/model/Recipe.ts
import {Cocktail} from "./Cocktail";
import {Ingredient} from "./Ingredient";
import { Exclude } from 'class-transformer';


export class Recipe{
  recipeId: number;

  ingredient: Ingredient;
  quantity: number;


  constructor(recipeId?: number, ingredient?: Ingredient, quantity?: number) {
    this.recipeId = recipeId;
    this.ingredient = ingredient;
    this.quantity = quantity;
  }

  // toJSON(): any {
  //   const { recipeId, ...rest } = this;
  //   return {
  //     '@class': 'com.gbyzzz.bar_web_app.bar_backend.dto.RecipeDTO',
  //     ...rest
  //   };
  // }
}

File: bar_backend/src/main/java/com/gbyzzz/bar_web_app/bar_backend/entity/Recipe.java
package com.gbyzzz.bar_web_app.bar_backend.entity;

import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import lombok.*;

import java.util.Objects;

@Entity
@Getter
@Setter
@ToString
@Builder
@AllArgsConstructor
@NoArgsConstructor
@Table(name = "recipes",
        uniqueConstraints=
        @UniqueConstraint(
                columnNames={"cocktail_id", "ingredient_id"}))
public class Recipe {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "recipe_id")
    private Long recipeId;

    @JsonIgnore
    @ManyToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "cocktail_id")
    private Cocktail cocktail;

    @ManyToOne(cascade = {CascadeType.MERGE, CascadeType.REFRESH, CascadeType.DETACH})
    @JoinColumn(name = "ingredient_id", nullable = false)
    private Ingredient ingredient;

    @Basic
    @Column(name = "quantity_of_ingredient")
    private Integer quantity;
}
